src/Shouldly.Shared/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs

```diff
@@ -48,6 +48,26 @@ public static void ShouldContain<T>(this IEnumerable<T> actual, [InstantHandle]
             ShouldContain(actual, elementPredicate, () => null);
         }
 
+        public static void ShouldContain<T>(this IEnumerable<T> actual, [InstantHandle] Expression<Func<T, bool>> elementPredicate, int expectedCount)
+        {
+            var condition = elementPredicate.Compile();
+            var actualCount = actual.Count(condition);
+            if (actualCount != expectedCount)
+            {
+                throw new ShouldAssertException(new ExpectedActualShouldlyMessage(expectedCount, actual, () => null).ToString());
+            }
+        }
+
+        public static void ShouldContain<T>(this IEnumerable<T> actual, [InstantHandle] Expression<Func<T, bool>> elementPredicate, int expectedCount, string customMessage)
```

Comment:
I think the `Func<string> customMessage` overload is missing.

Also you can make each of these methods call each other.